unit-tests/shared/src/test/scala/org/scalanative/testsuite/javalib/net/HttpCookieTest.scala

```diff
@@ -0,0 +1,73 @@
+package org.scalanative.testsuite.javalib.net
+
+import java.net._
+
+import org.junit.Test
+import org.junit.Assert._
+
+import org.scalanative.testsuite.utils.AssertThrows.assertThrows
+
+class HttpCookieTest {
+
+  @Test
+  def basicParseTest(): Unit = {
+
+    val out = HttpCookie.parse("set-cookie2:potato=tomato")
+    assertEquals(out.size(), 1)
+    val cookie = out.get(0)
+    assertEquals("potato", cookie.getName())
+    assertEquals("tomato", cookie.getValue())
+    assertEquals(null, cookie.getComment())
+    assertEquals(false, cookie.getDiscard())
+    assertEquals(null, cookie.getCommentURL())
+    assertEquals(null, cookie.getDomain())
+    assertEquals(-1, cookie.getMaxAge())
+    assertEquals(null, cookie.getPath())
+    assertEquals(null, cookie.getPortlist())
+    assertEquals(false, cookie.getSecure())
+    assertEquals(1, cookie.getVersion)
+
+  }
+
+  @Test
+  def attributeTest(): Unit = {
+    val out = HttpCookie.parse("""set-cookie2: 
+                               |potato = tomato ; 
+                               |comment = "This is a comment"; 
+                               |discard;
+                               |commenturl= https://definitely.real.website.cool/;
+                               |domain=definitely.real.webiste.cool;
+                               |max-age=123;
+                               |path=/potato/tomato;
+                               |port="123 456 789";
+                               |secure;
+                               |version=1""".stripMargin.replaceAll("\n", ""))
+
+    assertEquals(out.size(), 1)
+    val cookie = out.get(0)
+    assertEquals("potato", cookie.getName())
+    assertEquals("tomato", cookie.getValue())
+    assertEquals("This is a comment", cookie.getComment())
+    assertEquals(true, cookie.getDiscard())
+    assertEquals(
+      "https://definitely.real.website.cool/",
+      cookie.getCommentURL()
+    )
+    assertEquals("definitely.real.webiste.cool", cookie.getDomain())
+    assertEquals(123, cookie.getMaxAge())
+    assertEquals("/potato/tomato", cookie.getPath())
+    assertEquals("123 456 789", cookie.getPortlist())
+    assertEquals(true, cookie.getSecure())
+    assertEquals(1, cookie.getVersion)
+
+  }
+
+  @Test
+  def domainMatchTest: Unit = {
+    assert(HttpCookie.domainMatches("website.com", "website.com"))
+    assert(HttpCookie.domainMatches("website.com", "www.website.com"))
+    assert(!HttpCookie.domainMatches("www.website.com", "website.com"))
+    assert(!HttpCookie.domainMatches("website.com", "awebsite.com"))
+  }
+
+}
```

Comment:
One can spend eternities one does not have writing tests
for failure cases.  Users are _so_ good at generating just
the failure case not detected by tests.  Such is life.

The code goes to some length to detect`IllegalArgumentException("Illegal cookie name")`.

Did I miss a Test for that?  If not, you might want to consider adding it.

(is the "Illegal" in the message from JVM? If not, could it be changed
to something which does not keep me up all day wondering if the Sheriff's Deputy is coming for me because I typo-ed a name)

Reply:
> is the "Illegal" in the message from JVM?

Yes, I tried to mimic the behavior of the JVM; at least to a reasonable degree. Added tests for that now as well